### PR TITLE
For #17044: Explicitly set a new default engine when default is deleted.

### PR DESCRIPTION
**Pull request by mcarare**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Review by Amejia481 — CHANGES_REQUESTED** [review]
Looks great ✨ , could we add some tests to verify this behavior and avoid future regressions?

**Comment by mcarare** [comment]
@Amejia481 I also added some tests.

**Review by Amejia481 — APPROVED** [review]
Just one nit, after that is good to ship 🚢 !

File: app/src/test/java/org/mozilla/fenix/settings/search/RadioSearchEngineListPreferenceTest.kt [hunk 1]
```diff
@@ -0,0 +1,92 @@
+package org.mozilla.fenix.settings.search
```
[the remark this answers is not in the document]
Comment (mcarare) [reply app/src/test/java/org/mozilla/fenix/settings/search/RadioSearchEngineListPreferenceTest.kt:1]: Forgot about that. I was almost sure there was a lint for the license. 🤔

[the remark this answers is not in the document]
Comment (Amejia481) [reply app/src/test/java/org/mozilla/fenix/settings/search/RadioSearchEngineListPreferenceTest.kt:5]: Yes, I thought we have a linter too, we could open a bug to investigate, maybe is not running for some reasons

**Comment by pocmo** [comment]
@mcarare Can we please revert this and discuss this? The use case is for explicitly recording a user selection (and is getting persisted). In this case the user did not make this selection. We only persist user choices to have the ability to switch the default if it was *not* a user choice.

**Pull request merged**